Change some messages and documentation to Welkin, as part of our rebranding

cristiklein opened the pull request:
<!-- Choose your PR title carefully as it will be used as the entry in the changelog! -->

<!-- markdownlint-disable MD041 -->
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [x] personal data beyond what is necessary for interacting with this pull request, nor
> - [x] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       <!-- This PR adds a new feature -->
- [x] kind/improvement   <!-- This PR changes an existing feature -->
- [ ] kind/deprecation   <!-- This PR removes an existing feature -->
- [ ] kind/documentation <!-- This PR contains documentation -->
- [ ] kind/clean-up      <!-- This PR cleans up technical debt -->
- [ ] kind/bug           <!-- This PR fixes a bug -->
- [ ] kind/other         <!-- This PR does something else -->

_Optional_: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change   <!-- This PR introduces an admin facing change, add "Platform Administrator notice" section -->
- [ ] kind/dev-change     <!-- This PR introduces a dev facing change, add "Application Developer notice" section -->
- [ ] kind/security       <!-- This PR introduces a critical security fix, add "Security notice" section -->
- [ ] [kind/adr](set-me\) <!-- This PR implements an ADR, add the link -->

<!-- Uncomment the additional sections that applies. -->

### Release notes
Change some messages and documentation to Welkin, as part of our rebranding

<!-- Additional information with kind/admin-change
### Platform Administrator notice
...
-->

<!-- Add additional information with kind/dev-change
### Application Developer notice
...
-->

<!-- Add additional information with kind/security
### Security notice
...
-->

### What does this PR do / why do we need this PR?

<!-- Add description of the change -->

The purpose of this PR is for me to test how simple it is to use the instructions in [DEVELOPMENT.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/DEVELOPMENT.md) to make a simple improvement to Welkin.

As a side-effect, I decided to actually contribute. :smile:

<!-- Add all issues that are fixed by this PR, use "Part of" instead of "Fixes" if you want to keep issues open. -->
- Fixes #

#### Information to reviewers

![image](https://github.com/user-attachments/assets/5812a66a-a41d-4974-8d24-39a6a2a9e8ce)

#### Checklist

<!-- This section is not added to the changelog or release notes, it is to help you as a contributor and reviewers. -->

- [x] Proper commit message prefix on all commits
    <!-- Example of commit message prefixes:
    - all: changes to multiple areas
    - apps: changes to applications running in all clusters
    - apps sc: changes to applications running in service clusters
    - apps wc: changes to applications running in workload clusters
    - bin: changes to management binaries
    - config: changes to configuration
    - docs: changes to documentation
    - release: release related
    - scripts: changes to scripts
    - tests: changes to tests
    --->
- Change checks:
    - [] The change is transparent
    - [ ] The change is disruptive
    - [x] The change requires no migration steps
    - [ ] The change requires migration steps
    - [ ] The change updates CRDs
    - [ ] The change updates the config _and_ the schema
- Documentation checks:
    - [x] The [public documentation](https://github.com/elastisys/welkin) required no updates
    - [ ] The [public documentation](https://github.com/elastisys/welkin) required an update - [link to change](set-me\)
- Metrics checks:
    - [ ] The metrics are still exposed and present in Grafana after the change
    - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts required no updates)
    - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts required an update)
- Logs checks:
    - [ ] The logs do not show any errors after the change
- PodSecurityPolicy checks:
    - [ ] Any changed Pod is covered by Kubernetes Pod Security Standards
    - [ ] Any changed Pod is covered by Gatekeeper Pod Security Policies
    - [ ] The change does not cause any Pods to be blocked by Pod Security Standards or Policies
- NetworkPolicy checks:
    - [ ] Any changed Pod is covered by Network Policies
    - [ ] The change does not cause any dropped packets in the NetworkPolicy Dashboard
- Audit checks:
    - [ ] The change does not cause any unnecessary Kubernetes audit events
    - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
    - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
    - [ ] The bug fix is covered by regression tests

llarsson reviewed (commented):
Good changes so stuff works without redirects! 

I see the addition of the Registered Trademark symbol. If we use that, we ought to ensure that we always write it as "Elastisys Welkin®", because our registered trademark is actually for those two words -- the _Elastisys_ part is required as well.

Edit: is it possible to set something like that up using the "vale" or whatever the not-a-spellchecker-but-almost tool is called?

cristiklein commented:
> Good changes so stuff works without redirects!

Thanks!

> I see the addition of the Registered Trademark symbol. If we use that, we ought to ensure that we always write it as "Elastisys Welkin®", because our registered trademark is actually for those two words -- the _Elastisys_ part is required as well.

Nice catch. Fixed!

> Edit: is it possible to set something like that up using the "vale" or whatever the not-a-spellchecker-but-almost tool is called?

In my experience, Vale is rather bad at "multi-word" spellchecking. I added a custom pre-commit hook.

linus-astrom reviewed (approved):
LGTM

aarnq commented on .pre-commit-config.yaml:
```diff
@@ -1,4 +1,23 @@
+        language: pygrep
+        files: ".*[.]md$"
+        exclude: ^changelog/0.43.md$
+      - id: whole-trademark
+        name: Use Elastisys Welkin® consistently
+        entry: python ./scripts/check-elastisys-welkin.py
+        language: python
+        types: [text]
```
```suggestion
      - id: consistent-trademark
        name: Either use the trademark "Elastisys Welkin®" consistently or "Welkin" on its own
        entry: "(?<!Elastisys )\\bWelkin®"
        language: pygrep
        types: [text]
```
This doesn't need a script?

cristiklein replied:
Double-backslash! That's what I missed, thanks!

aarnq commented on .pre-commit-config.yaml:
```diff
@@ -1,4 +1,23 @@
 repos:
+  - repo: local
+    hooks:
+      - id: avoid-old-website-urls
+        name: Avoid old URLs (https://elastisys.io/compliantkubernetes)
+        entry: "elastisys.io/compliantkubernetes"
+        language: pygrep
+        files: ".*[.]md$"
```
```suggestion
      - id: avoid-old-website-urls
        name: Avoid old URLs (https://elastisys.io/compliantkubernetes)
        entry: "elastisys.io/compliantkubernetes"
        language: pygrep
        types: [text]
        exclude: ^.pre-commit-config.yaml$
```
We have links in more files than just markdown.

aarnq commented on .pre-commit-config.yaml:
```diff
@@ -1,4 +1,23 @@
+        language: pygrep
+        files: ".*[.]md$"
+      - id: avoid-old-names
+        name: Use Welkin consistently
```
```suggestion
        name: Avoid old names and use use "Welkin" instead
```
To be consistent with the other one.

cristiklein commented on .pre-commit-config.yaml:
```diff
@@ -1,4 +1,23 @@
 repos:
+  - repo: local
+    hooks:
+      - id: avoid-old-website-urls
+        name: Avoid old URLs (https://elastisys.io/compliantkubernetes)
+        entry: "elastisys.io/compliantkubernetes"
+        language: pygrep
+        files: ".*[.]md$"
```
Good point! The pre-commit message should focus on "how to do it right" and not "this is what you did wrong". Thanks!

llarsson reviewed (approved):
Nice, a lot of really good changes that are also visible to Application Developers! Love it!
